contracts/Overtime/SportsAMMV2RiskManager.sol

```diff
@@ -127,6 +130,30 @@ contract SportsAMMV2RiskManager is Initializable, ProxyOwned, ProxyPausable, Pro
         return _totalSpent <= capToBeUsed * riskMultiplier;
     }
 
+    /// @notice returns if parlay has any illegal combinations
+    /// @param _tradeData the parlay to check
+    function hasIllegalCombinationsOnTicket(ISportsAMMV2.TradeData[] memory _tradeData) external view returns (bool) {
+        for (uint i = 0; i < _tradeData.length; i++) {
+            ISportsAMMV2.TradeData memory tradeDataItemCurrent = _tradeData[i];
+            for (uint j = i + 1; j < _tradeData.length; j++) {
+                ISportsAMMV2.TradeData memory tradeDataItemToCheckAgainst = _tradeData[j];
+                if (
+                    tradeDataItemCurrent.gameId == tradeDataItemToCheckAgainst.gameId &&
+                    tradeDataItemCurrent.sportId == tradeDataItemToCheckAgainst.sportId
+                ) {
+                    if (!combiningPerSportEnabled[tradeDataItemCurrent.sportId]) {
```

Comment:
Whole logic can go under one `if` with `||`, but probably compiled the same in the background.